core/dbt/context/base.py

```diff
@@ -168,7 +170,11 @@ def node_name(self) -> str:
         else:
             return "<Configuration>"
 
-    def get_missing_var(self, var_name: str) -> NoReturn:
+    def get_missing_var(self, var_name: str):
+        # If vars are not required (e.g., during deps), return None
+        # instead of raising an error
+        if not self._require_vars:
+            return None
         # TODO function name implies a non exception resolution
         raise RequiredVarNotFoundError(var_name, dict(self._merged), self._node)
```

Comment:
A simpler implementation could be
```suggestion
    def get_missing_var(self, var_name: str) -> None:
        # Only raise an error if vars are _required_
        if self._require_vars:
          raise RequiredVarNotFoundError(var_name, dict(self._merged), self._node)
```